Model/Api/AdyenDonations.php

```diff
@@ -96,6 +107,32 @@ public function donate($payload)
         }
 
         $donationsCaptureCommand = $this->commandPool->get('capture');
-        return $donationsCaptureCommand->execute(['payment' => $payload]);
+        $donationResult = $donationsCaptureCommand->execute(['payment' => $payload]);
+
+        // Remove donation token after a successfull donation.
+        $this->removeDonationToken($order);
+
+        return $donationResult;
+    }
+
+    private function incrementTryCount($order)
+    {
+        $donationTryCount = $order->getPayment()->getAdditionalInformation('donationTryCount');
+
+        if (!$donationTryCount) {
+            $order->getPayment()->setAdditionalInformation('donationTryCount', 1);
+            $order->save();
```

Comment:
save can be called for both IF and ELSE if moved after the closing bracket

Reply:
https://github.com/Adyen/adyen-magento2/pull/1428/commits/b6068376284c73fd88555611dc96e08375aba649